File: day_15.py
```
"""

https://www.codechef.com/problems/BUY1GET1

"""
import math
t = int(input())
for _ in range(t):
    s = input()
    d = {}
    for i in range(len(s)):
        d[s[i]] = 0
    for i in range(len(s)):
        d[s[i]] += 1
    count = 0
    cost = 0
    for key in d:
        cost = math.ceil(d[key] / 2)
        count += cost
    print(count)

# sssssssaaa
# {'s' : 7, 'a': 3}
#  7/2
```

File: lazydev01/day_15.cpp
/*

Question -> https://www.codechef.com/problems/BUY1GET1

*/
#include <bits/stdc++.h>
using namespace std;

#ifdef ONLINE_JUDGE
  #define cerr if(0) cerr
#endif

#define int long long int
#define endl "\n"
#define YES cout<<"YES\n"
#define NO cout<<"NO\n"

int32_t main()
{
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    int t;
    cin >> t;

    while (t--)
    {
        string str;
        cin >> str;
        map<char, int> m;
        for(int i=0; i<str.size(); i++){
            m[str[i]]++;
        }
        int count = 0;
        for(auto i: m){
            count+=((i.second+1)/2);
        }
        cout << count << endl;
    }
    return 0;
}